src/components/IngredientSearch/IngredientSearch.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext, useEffect, useState } from "react";
+import { AppContext } from "../AppContext/AppContext";
+import { IngredientDisplay } from "../IngredientDisplay/IngredientDisplay";
+
+const SearchStyles: React.CSSProperties = {
+  fontSize: "1.5rem",
+  padding: "1rem",
+  width: "50%",
+};
+
+export const IngredientSearch = () => {
+  const { data } = useContext(AppContext);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [matchingImages, setMatchingImages] = useState([]);
+
+  useEffect(() => {
```

Comment:
This is my not fully understanding React/the difference between the two - would `useMemo` be better here than `useEffect`?

Reply:
Also should we start using hooks with this?